tests/test_query.py

```diff
@@ -661,3 +669,67 @@ def test_integration_query_without_auth_succeeds(
 
     response = test_app.get(ROUTE)
     assert response.status_code == 200
+
+
+def test_derivatives_info_handled_by_agg_api_response(
+    test_app,
+    mock_post_agg_query_to_graph,
+    mock_query_matching_dataset_sizes,
+    monkeypatch,
+    mock_auth_header,
+    set_mock_verify_token,
+):
+    """
+    Test that in the aggregated API mode, pipeline information for matching subjects
+    is correctly parsed and formatted in the API response.
+    """
+    monkeypatch.setattr(
+        util, "RETURN_AGG", util.EnvVar(util.RETURN_AGG.name, True)
+    )
+    monkeypatch.setattr(
+        crud, "post_query_to_graph", mock_post_agg_query_to_graph
+    )
+    monkeypatch.setattr(
+        crud, "query_matching_dataset_sizes", mock_query_matching_dataset_sizes
+    )
+
+    response = test_app.get(ROUTE, headers=mock_auth_header)
+    assert response.status_code == 200
+
+    matching_ds = response.json()[0]
+    assert matching_ds["available_pipelines"] == {
+        "https://github.com/nipoppy/pipeline-catalog/tree/main/processing/freesurfer": [
+            "7.3.2"
+        ]
+    }
+
+
+def test_missing_derivatives_info_handled_by_nonagg_api_response(
+    test_app,
+    mock_post_nonagg_query_to_graph,
+    mock_query_matching_dataset_sizes,
+    monkeypatch,
+    mock_auth_header,
+    set_mock_verify_token,
+):
+    """
+    Test that in the non-aggregated API mode, when all matching subjects lack pipeline information,
```

Comment:
Maybe I'm missing it, but where is the test that

- has no pipeline info in the graph
- is running API in aggregated mode
- does not fail

? Because afaiu this is the test we would expect to catch our bug right now, yes?

Reply:
That test doesn't exist. Do you think it's necessary to add it in this PR?  It wouldn't catch the current bug, so I'm not sure it would be super relevant here - rather, I think it should be addressed as part of https://github.com/neurobagel/api/issues/371.

The test that _would_ fail under the original bug conditions is the one you highlighted here, which is newly added, i.e.:
https://github.com/neurobagel/api/blob/b1733213de58911285b5f5fa7174910435e5e888/tests/test_query.py#L707-L735

Reply:
aaah, thanks. I forgot that we had the bug when running in **non-agg** mode, not (as I falsely believed) in agg mode. So disregard my comment - agree with your answer